test/upgrader_test.dart

```diff
@@ -452,6 +452,19 @@ void main() {
     expect(called, false);
     expect(notCalled, true);
   });
+
+  test('test upgrader shouldDisplayUpgrade', () async {
+    final client = MockClient.setupMockClient();
+    final upgrader = Upgrader();
+    upgrader.client = client;
+    upgrader.debugLogging = true;
+
+    expect(upgrader.shouldDisplayUpgrade(), false);
+    upgrader.debugDisplayAlways = true;
+    expect(upgrader.shouldDisplayUpgrade(), true);
+    upgrader.debugDisplayAlways = false;
+    expect(upgrader.shouldDisplayUpgrade(), false);
+  });
```

Comment:
I've added a PR to increase the testing around the functionality this PR adds (#44).